packages/host/app/components/operator-mode.gts

```diff
@@ -1,42 +1,86 @@
 import Component from '@glimmer/component';
-import { Card, Format, CardInstanceType } from 'https://cardstack.com/base/card-api';
-import { tracked } from '@glimmer/tracking';
+import { Card, Format } from 'https://cardstack.com/base/card-api';
 import Preview from './preview';
 import { action } from '@ember/object';
 import { on } from '@ember/modifier';
 import { fn } from '@ember/helper';
-import { service } from '@ember/service';
+import { Button } from '@cardstack/boxel-ui';
+import CardCatalogModal from '@cardstack/host/components/card-catalog-modal';
+import CreateCardModal from '@cardstack/host/components/create-card-modal';
 import type CardService from '../services/card-service';
-import type RouterService from '@ember/routing/router-service';
 import getValueFromWeakMap from '../helpers/get-value-from-weakmap';
 import { eq, not } from '@cardstack/boxel-ui/helpers/truth-helpers';
 import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
 import cn from '@cardstack/boxel-ui/helpers/cn';
 import { restartableTask } from 'ember-concurrency';
-import { TrackedWeakMap } from 'tracked-built-ins';
-import { Model as CardRouteModel } from '../routes/card';
+import {
+  chooseCard,
+  catalogEntryRef,
+  createNewCard,
+  baseRealm,
+} from '@cardstack/runtime-common';
+import { CatalogEntry } from 'https://cardstack.com/base/catalog-entry';
+import type LoaderService from '../services/loader-service';
+import { service } from '@ember/service';
+import type * as CardAPI from 'https://cardstack.com/base/card-api';
+
+import { TrackedArray, TrackedWeakMap } from 'tracked-built-ins';
 
 interface Signature {
   Args: {
-    firstCardInStack: CardRouteModel;
+    firstCardInStack: Card;
   };
 }
 
 export default class OperatorMode extends Component<Signature> {
-  @tracked stack: Card[] = [];
+  stack: Card[];
   formats: WeakMap<Card, Format> = new TrackedWeakMap<Card, Format>();
 
   //A variable to store value of card field
   //before in edit mode.
   cardFieldValues: WeakMap<Card, Map<string, any>> = new WeakMap<Card, Map<string, any>>();
+  @service declare loaderService: LoaderService;
   @service declare cardService: CardService;
-  @service declare router: RouterService;
 
   constructor(owner: unknown, args: any) {
     super(owner, args);
-    this.stack = [this.args.firstCardInStack.card!];
+    this.stack = new TrackedArray([this.args.firstCardInStack]);
```

Comment:
perhaps we can make this.stack of type TrackedArray above? then it is more clear that it is a tracked property.